File: Python/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cnt = ans = nums[0]
        for i in range(1, len(nums)):
            if cnt < 0:
                cnt = 0
            cnt += nums[i]
            ans = max(ans, cnt)
        return ans

File: C++/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int len = nums.size();
        int sum = nums[0],ans = nums[0];
        for(int i=1;i<len;i++)
        {
            if(sum < 0)
                sum = 0;
            sum += nums[i];
            ans = max(ans,sum);
        }
        return ans;
    }
};